client/js/compute.js

```diff
@@ -0,0 +1,578 @@
+      sample: 'var a = ${x} - ${y}',
+      call: function(v, x, y) { return v = x - y }
+    },
+    {
+      name: '*',
+      desc: 'returns the product of two numbers (multiply x and y)',
+      sample: 'var a = ${x} * ${y}',
+      call: function(v, x, y) { return v = x * y }
+    },
+    {
+      name: '**',
+      desc: 'returns the result of raising x to the power of y (exponent)',
+      sample: 'var a = ${x} ** ${y}',
+      call: function(v, x, y) { return v = x ** y }
+    },
+    {
+      name: '/',
+      desc: 'returns the division of two numbers (divide x by y)',
+      sample: 'var a = ${x} / ${y}',
+      call: function(v, x, y) { return v = x / y }
+    },
+    {
+      name: '%',
+      desc: 'returns the integer remainder/rest/modal of the division of two numbers',
+      sample: 'var a = ${x} % ${y}',
+      call: function(v, x, y) { return v = x % y }
+    },
+    {
+      name: '<',
+      desc: 'returns true if x is less than y',
+      sample: 'var a = ${x} < ${y}',
+      call: function(v, x, y) { return v = x < y }
+    },
+    {
+      name: '<=',
+      desc: 'returns true if x is less than or equal to y',
+      sample: 'var a = ${x} <= ${y}',
+      call: function(v, x, y) { return v = x <= y }
+    },
+    {
+      name: '==',
+      desc: 'returns true if x is equal to y',
+      sample: 'var a = ${x} == ${y}',
+      call: function(v, x, y) { return v = x == y }
+    },
+    {
+      name: '!=',
+      desc: 'returns true if x is not equal to y',
+      sample: 'var a = ${x} != ${y}',
+      call: function(v, x, y) { return v = x != y }
+    },
+    {
+      name: '>=',
+      desc: 'returns true if x is greater than or equal to y',
+      sample: 'var a = ${x} >= ${y}',
+      call: function(v, x, y) { return v = x >= y }
+    },
+    {
+      name: '>',
+      desc: 'returns true if x is greater than y',
+      sample: 'var a = ${x} > ${y}',
+      call: function(v, x, y) { return v = x > y }
+    },
+    {
+      name: '&&',
+      desc: 'logical AND returns true if both x and y are true or returns the value of x or y if they are not boolean values',
+      sample: 'var a = ${x} >= ${y}',
+      call: function(v, x, y) { return v = x && y }
+    },
+    {
+      name: '||',
+      desc: 'logical OR returns true if either x or y are true or returns the value of x or y if they are not boolean values',
+      sample: 'var a = ${x} >= ${y}',
+      call: function(v, x, y) { return v = x || y }
+    },
+    {
+      name: '!',
+      desc: 'logical NOT returns true if the given value is false (or can be evaluated as false), otherwise returns false',
+      sample: 'var a = ! ${x}',
+      call: function(v, x) { return v = !x }
+    },
+    {
+      name: 'hypot',
+      desc: 'returns the square root of the sum of squares (hypotenuse) - a second number y is optional',
+      sample: 'var a = hypot ${x} ${y}',
+      call: function(v, x, y) { return v = (y !== undefined) ? Math.hypot(x, y) : Math.hypot(x) }
```

Comment:
On both servers it appears to behave the same.  However, null and zero are treated as the default operand of 1.  So the hypotenuse of 4 and null, or 4 and 0 will be returned as the square root of 17 instead of the square root of 16.  Maybe that is what he was trying to correct.  The COMPUTE tutorial does not show this error.  Maybe because it uses maththis.  However, a button (such as the one below) in the Tutorials room will show the error.
![image](https://user-images.githubusercontent.com/78324099/119266642-4075ce80-bbb1-11eb-8fe3-0f85763a608e.png)

Reply:
They appear to behave the same (correctly) on both servers.

Reply:
It doesn't change anything now because the `getValue` function currently doesn't allow for undefined y and z values (defaulting to 1), but that's actually not correct as `Math.hypot(2) != Math.hypot(2,1)`. I'm already preparing the compute operations to work as they should, while keeping everything working as-is for now.

Reply:
Looks like my assumption was not correct. `Math.hypot(1, undefined)` does not return the same thing as `Math.hypot(1)`. Weird.